### DISCUSSION TITLE
[iOS/IPadOS] Why does text containing ":" get url-encoded when pasted into editor?

### DISCUSSION THREAD

**Discussion by travisnburton** [coder/code-server#7057] [Q&A]
When pasting text where the first line contains `:` into an editor window, all text following the colon is url-encoded.

Example text without colon on first line:
```
text
encoding: none
foo bar baz
```

Normal result when pasted into editor:
```
text
encoding: none
foo bar baz
```

Example text with colon on first line: 
```
text:
encoding: url
foo bar baz
```

Encoded result when pasted into editor:
```
text:%0Aencoding%3A%20url%0Afoo%20bar%20baz
```

This encoding does not happen when pasting text into the code-server terminal, it only happens when pasting into code-server editor windows.

I have no extensions installed, and all settings are default.

**Comment by code-asher** [comment]
Interesting, any idea if it happens to you in Codespaces?  I ask because it seems most likely an upstream issue with VS Code itself, if I had to guess.  I tried to reproduce in code-server 4.93.1 in a brand new file called `new` but it worked as expected for me.

**Reply by code-asher** [reply]
And to test I used `ctrl+c` `ctrl+v` in Firefox.

**Reply by travisnburton** [reply]
It does seem to be a problem with VS Code, but i have no idea what it could be.  I cannot replicate it in Codespaces.

**Comment by travisnburton** [comment]
Apparently this is the way Apple has decided to treat strings containing a colon - https://stackoverflow.com/questions/77662557/how-do-you-avoid-copy-paste-content-with-a-coming-out-url-encoded-on-an-ipho

It's strange that i haven't been able to find any other conversations/issues about this here.

**Reply by code-asher** [reply]
Very interesting!!  Good to know, and nice find.

**Reply by code-asher** [reply]
Strange that it does not reproduce in Codespaces though, I wonder if they apply a workaround there.

**Comment by travisnburton** [comment]
Resolved by disabling `editor.pasteAs.enabled`
